src/coreclr/clr.featuredefines.props

```diff
@@ -60,7 +60,7 @@
         <DefineConstants Condition="'$(FeatureCominteropWinRTManagedActivation)' == 'true'">$(DefineConstants);FEATURE_COMINTEROP_WINRT_MANAGED_ACTIVATION</DefineConstants>
         <DefineConstants Condition="'$(FeatureManagedEtw)' == 'true'">$(DefineConstants);FEATURE_MANAGED_ETW</DefineConstants>
         <DefineConstants Condition="'$(FeatureManagedEtwChannels)' == 'true'">$(DefineConstants);FEATURE_MANAGED_ETW_CHANNELS</DefineConstants>
-        <DefineConstants Condition="'$(FeaturePal)' == 'true'">$(DefineConstants);FEATURE_PAL</DefineConstants>
+        <DefineConstants Condition="'$(FeaturePal)' == 'true'">$(DefineConstants);TARGET_UNIX</DefineConstants>
```

Comment:
The CPU architecture defines have underscore around them. This one does not. It would be nice to be consistent (it will be a bug farm otherwise).

Reply:
Not a good idea.  Identifiers starting with underscore are reserved for the standard, at least in C.  It's undefined behavior to define them yourself.  (The CPU architecture defines are like that because they're provided by the compiler.)

Reply:
@lperiera For consistency it makes sense.  We use that style a lot currently.  If we want to change it should be a separate PR.

For the Preprocessor it is common practice to make file #ifndef to use a form `_Name_H_` or something like that.  Certainly we could collide with a compile #define, but it is unlikely.
 

```
$ git grep add_definitions | grep TARGET_
clrdefinitions.cmake:    add_definitions(-DDBG_TARGET_AMD64_UNIX)
clrdefinitions.cmake:  add_definitions(-D_TARGET_AMD64_)
clrdefinitions.cmake:  add_definitions(-D_TARGET_64BIT_)
clrdefinitions.cmake:  add_definitions(-DDBG_TARGET_64BIT)
clrdefinitions.cmake:  add_definitions(-DDBG_TARGET_AMD64)
clrdefinitions.cmake:    add_definitions(-DDBG_TARGET_ARM64_UNIX)
clrdefinitions.cmake:  add_definitions(-D_TARGET_ARM64_)
clrdefinitions.cmake:  add_definitions(-D_TARGET_64BIT_)
clrdefinitions.cmake:  add_definitions(-DDBG_TARGET_64BIT)
clrdefinitions.cmake:  add_definitions(-DDBG_TARGET_ARM64)
clrdefinitions.cmake:    add_definitions(-DDBG_TARGET_ARM_UNIX)
clrdefinitions.cmake:  add_definitions(-D_TARGET_ARM_)
clrdefinitions.cmake:  add_definitions(-DDBG_TARGET_32BIT)
clrdefinitions.cmake:  add_definitions(-DDBG_TARGET_ARM)
clrdefinitions.cmake:  add_definitions(-D_TARGET_X86_)
clrdefinitions.cmake:  add_definitions(-DDBG_TARGET_32BIT)
clrdefinitions.cmake:  add_definitions(-DDBG_TARGET_X86)
clrdefinitions.cmake:  add_definitions(-D_TARGET_UNIX_)
clrdefinitions.cmake:  add_definitions(-D_TARGET_WINDOWS_)
src/debug/debug-pal/CMakeLists.txt:  add_definitions(-D_TARGET_UNIX_)
src/debug/debug-pal/CMakeLists.txt:  add_definitions(-D_TARGET_WINDOWS_)
src/gcinfo/gcinfo_arm/CMakeLists.txt:add_definitions(-D_TARGET_ARM_)
src/gcinfo/gcinfo_arm64/CMakeLists.txt:add_definitions(-D_TARGET_ARM64_)
src/jit/armelnonjit/CMakeLists.txt:    add_definitions(-D_TARGET_ARM_)
src/jit/armelnonjit/CMakeLists.txt:    add_definitions(-D_TARGET_ARM64_)
src/jit/protononjit/CMakeLists.txt:    add_definitions(-D_TARGET_ARM_)
src/jit/protononjit/CMakeLists.txt:    add_definitions(-D_TARGET_ARM64_)
src/pal/src/CMakeLists.txt:  add_definitions(-D_TARGET_MAC64)
src/pal/src/CMakeLists.txt:add_definitions(-D_TARGET_UNIX_)
src/pal/tests/CMakeLists.txt:add_definitions(-D_TARGET_UNIX_)
tests/src/Interop/PInvoke/Generics/CMakeLists.txt:  add_definitions(-D_TARGET_X86_)
tests/src/Interop/PInvoke/Generics/CMakeLists.txt:  add_definitions(-D_TARGET_XARCH_)
tests/src/Interop/PInvoke/Generics/CMakeLists.txt:  add_definitions(-D_TARGET_AMD64_)
tests/src/Interop/PInvoke/Generics/CMakeLists.txt:  add_definitions(-D_TARGET_XARCH_)
tests/src/Interop/PInvoke/Generics/CMakeLists.txt:  add_definitions(-D_TARGET_ARM_)
tests/src/Interop/PInvoke/Generics/CMakeLists.txt:  add_definitions(-D_TARGET_ARMARCH_)
tests/src/Interop/PInvoke/Generics/CMakeLists.txt:  add_definitions(-D_TARGET_ARM64_)
tests/src/Interop/PInvoke/Generics/CMakeLists.txt:  add_definitions(-D_TARGET_ARMARCH_)
```

Reply:
I am fine with either style - `_TARGET_XXX_` or `TARGET_XXX` - as long as it is used for all `XXX`.